aspnetcore/fundamentals/configuration.md

```diff
@@ -1,166 +1,357 @@
 
-## Getting and setting configuration settings
+  ## Simple configuration
 
-ASP.NET Core's configuration system has been re-architected from previous versions of ASP.NET, which relied on `System.Configuration` and XML configuration files like `web.config`. The new configuration model provides streamlined access to key/value based settings that can be retrieved from a variety of sources. Applications and frameworks can then access configured settings in a strongly typed fashion using the new [Options pattern](xref:fundamentals/configuration#options-config-objects).
+The following console app uses the JSON configuration provider:
 
-To work with settings in your ASP.NET application, it is recommended that you only instantiate a `Configuration` in your application's `Startup` class. Then, use the [Options pattern](xref:fundamentals/configuration#options-config-objects) to access individual settings.
+[!code-csharp[Main](configuration/sample/src/ConfigJson/Program.cs)]
 
-At its simplest, `Configuration` is just a collection of sources, which provide the ability to read and write name/value pairs. If a name/value pair is written to `Configuration`, it is not persisted. This means that the written value will be lost when the sources are read again.
+The app reads and displays the following configuration settings:
 
-You must configure at least one source in order for `Configuration` to function correctly. The following sample shows how to test working with `Configuration` as a key/value store:
+[!code-json[Main](configuration/sample/src/ConfigJson/appsettings.json)]
 
-[!code-csharp[Main](configuration/sample/src/CodeSnippets/ConfigSummarySnippet.cs?range=11-18)]
+Configuration consists of a hierarchical list of name-value pairs in which the nodes are separated by a colon. To retrieve a particular value, you access the `Configuration` indexer with the corresponding item’s key:
 
-> [!NOTE]
-> You must set at least one configuration source.
+```
+   Console.WriteLine(
+     $"option1 = {Configuration["subsection:suboption1"]}");
+   ```
 
-It's not unusual to store configuration values in a hierarchical structure, especially when using external files (e.g. JSON, XML, INI). In this case, configuration values can be retrieved using a `:` separated key, starting from the root of the hierarchy. For example, consider the following *appsettings.json* file:
+Name/value pairs written to the built in `Configuration` providers are **not** persisted, however, you can create a custom provider that saves values. See [custom configuration provider](xref:fundamentals/configuration#custom-config-providers).
 
-<a name=config-json></a>
+The sample above uses the configuration indexer to read values. In ASP.NET Core applications, we recommend you use the [options pattern](xref:fundamentals/configuration#options-config-objects) rather than the indexer to read configuration values. We'll show that later in this document.
```

Comment:
I think the recommendation to use Options can be made more specific to how to flow information from configuration into other parts of the application outside of Startup.cs. Accessing values using the indexer where the Configuration object is available sound ok to me.